crates/ruma-common/src/identifiers/key_id.rs

```diff
@@ -33,12 +66,30 @@ impl<A: KeyAlgorithm, K: KeyName + ?Sized> KeyId<A, K> {
         Self::from_borrowed(&res).to_owned()
     }
 
-    /// Returns key algorithm of the key ID.
+    /// Returns key algorithm of the key ID - the part that comes before the colon.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use ruma_common::{KeyId, OwnedDeviceKeyId};
+    ///
+    /// let k: OwnedDeviceKeyId = KeyId::parse("ed25519:1").unwrap();
+    /// assert_eq!(k.algorithm().as_str(), "ed25519");
+    /// ```
     pub fn algorithm(&self) -> A {
         A::from(&self.as_str()[..self.colon_idx()])
     }
 
-    /// Returns the key name of the key ID.
+    /// Returns the key name of the key ID - the part that comes after the colon.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use ruma_common::{KeyId, OwnedDeviceKeyId};
+    ///
+    /// let k: OwnedDeviceKeyId = KeyId::parse("ed25519:foo").unwrap();
+    /// assert_eq!(k.key_name(), "foo");
```

Comment:
To show that the string is typed, we should use the `device_id!` macro on the right side.

Reply:
Done in 23854e6f